microsoft-edge/webview2/how-to/webresourcerequested.md

```diff
@@ -0,0 +1,582 @@
+---
+title: Custom management of network requests
+description: Using the WebResourceRequested event and the WebResourceResponseReceived event in WebView2 apps.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 04/29/2022
+---
+# Custom management of network requests
+<!--
+# Custom management of network requests and responses in WebView2
+# Managing network requests in WebView2
+# Navigating with web resource request and response events
+-->
+
+<!-- todo
+writer: add arrows to diagrams
+writer: read through body content
+-->
+
+The Microsoft Edge WebView2 control lets you interact with and modify network requests.  You can either provide a response or modify the network request using the `webResourceRequested` and `webResourceResponseReceived` events.  There is also special functionality that allows you to navigate with specific network requests using the `NavigateWithWebResourceRequest API`.
+
+This article describes how you can modify network requests.  Use this API and approach to:
+* Upload local file content to your app to add support for offline functionality.
+* Block content in a webpage, such as specific images.
+* Fine-tune authentication for specific pages.
+
+**Terminology:**
+
+| Term | Definition |
+|---|---|
+| intercept | Your host app can intercept (receive) a request that is sent from the WebView2 control to the HTTP server, read or modify the request, and then send the unchanged or modified request to the HTTP server (or to local code instead of the HTTP server). |

```

Comment:
The host app doesn't 'receive' the request, it intercepts it before it's sent to the server. I think receive is the wrong word here and only would cause confusion.